src/main/java/com/google/googleidentity/oauth2/validator/AuthorizationEndpointRequestValidator.java

```diff
@@ -106,7 +106,18 @@ public static void validateGET(
             throw new UnsupportedResponseTypeException();
         }
 
-        String grantType = OAuth2Utils.getGrantTypeFromResponseType(responseType);
+        ClientDetails.GrantType grantType = ClientDetails.GrantType.NONE;
+
+        switch (OAuth2Utils.getGrantTypeFromResponseType(responseType)) {
+            case OAuth2Constants.GrantType.AUTHORIZATION_CODE:
+                grantType = ClientDetails.GrantType.AUTHORIZATION_CODE;
+                break;
+            case OAuth2Constants.GrantType.IMPLICIT:
+                grantType = ClientDetails.GrantType.IMPLICIT;
+            default:
+                // Never happen
```

Comment:
Things that should never happen should throw IllegalStateException, either directly or via Preconditions.checkState calls. Otherwise, if there is a bug, this code will silently continue rather than telling you something bad is happening.